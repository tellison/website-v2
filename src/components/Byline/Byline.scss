/* Mobile specific css */
@media (max-width: 650px) {
  .byline {
    font-size: 0.7em !important;
  }
}
