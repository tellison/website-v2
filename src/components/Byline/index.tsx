import React from 'react';
import { Link } from 'gatsby-plugin-react-i18next';
import ProfilePicInline from '../ProfilePicInline';
import './Byline.scss';

const Byline = (props) => {
  const { author, date, identifier } = props;

  const href = `/blog/author/${identifier}`;

  return (
    <div
      style={{
        display: 'flex',
        alignItems: 'center',
        marginTop: '1rem',
        marginBottom: '1rem',
        textDecoration: 'none'
      }}
      className='byline'
    >
      {date} â€“ posted by &nbsp; <Link to={href}>{author}</Link> <ProfilePicInline identifier={identifier} name={author} />
    </div>
  );
};

export default Byline;