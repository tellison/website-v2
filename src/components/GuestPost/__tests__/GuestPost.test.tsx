import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import Byline from '../index';

describe('Byline component', () => {
    it('should render correctly', () => {
        const { container } = render(
            <Byline
                children={[]}
            />
        );

        expect(container).toMatchSnapshot();
    });
});