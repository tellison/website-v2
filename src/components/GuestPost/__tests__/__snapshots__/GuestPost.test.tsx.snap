// Vitest Snapshot v1

exports[`Byline component > should render correctly 1`] = `
<div>
  <p
    style="padding: 0.5em; background-color: rgb(227, 227, 227);"
  >
    <em>
       â€“ Adoptium Team
    </em>
  </p>
</div>
`;
